tests/api/core/strapi/api/sort/sort.test.api.ts

```diff
@@ -0,0 +1,403 @@
+      { id: 1, title: 'Article A', ...BASIC_ARTICLE_FIELDS, categories: [cat('b')] },
+    ]);
+  });
+
+  test('Deep sort (2st level) + Pagination (start/limit)', async () => {
+    const res = await rq.get(`/${schemas.contentTypes.article.pluralName}`, {
+      qs: {
+        sort: 'categories.tags.name',
+        pagination: { start: 1, limit: 2 },
+        populate: 'categories.tags',
+      },
+    });
+
+    expect(res.status).toBe(200);
+    expect(res.body.data.length).toBe(2);
+
+    expect(res.body.data).toMatchObject([
+      { id: 3, title: 'Article D', ...BASIC_ARTICLE_FIELDS, categories: [cat('a'), cat('d')] },
+      { id: 4, title: 'Article B', ...BASIC_ARTICLE_FIELDS, categories: [cat('c'), cat('d')] },
+    ]);
+  });
+
+  test('Deep sort (1st level) + Pagination (page/pageSize)', async () => {
+    const pageSize = 3;
+
+    const page1 = await rq.get(`/${schemas.contentTypes.article.pluralName}`, {
+      qs: { sort: 'categories.name', populate: 'categories', pagination: { pageSize, page: 1 } },
+    });
+
+    const page2 = await rq.get(`/${schemas.contentTypes.article.pluralName}`, {
+      qs: { sort: 'categories.name', populate: 'categories', pagination: { pageSize, page: 2 } },
+    });
+
+    expect(page1.status).toBe(200);
+    expect(page1.body.data.length).toBe(3);
+
+    expect(page2.status).toBe(200);
+    expect(page2.body.data.length).toBe(1);
+
+    // Page 1
+    expect(page1.body.data).toMatchObject([
+      { id: 3, ...BASIC_ARTICLE_FIELDS, title: 'Article D', categories: [cat('a'), cat('d')] },
+      { id: 1, ...BASIC_ARTICLE_FIELDS, title: 'Article A', categories: [cat('b')] },
+      { id: 4, ...BASIC_ARTICLE_FIELDS, title: 'Article B', categories: [cat('c'), cat('d')] },
+    ]);
+
+    expect(page1.body.meta).toMatchObject({
+      pagination: { page: 1, pageSize, pageCount: 2, total: 4 },
+    });
+
+    // Page 2
+    expect(page2.body.data).toMatchObject([
+      { id: 2, ...BASIC_ARTICLE_FIELDS, title: 'Article C', categories: [cat('d')] },
+    ]);
+
+    expect(page2.body.meta).toMatchObject({
+      pagination: { page: 2, pageSize, pageCount: 2, total: 4 },
+    });
+  });
+
+  test('Deep sort (1st level) + Filters', async () => {
+    const res = await rq.get(`/${schemas.contentTypes.article.pluralName}`, {
+      qs: {
+        sort: ['categories.name:ASC', 'title:DESC'],
+        filters: { title: { $ne: 'Article B' } },
+      },
+    });
+
+    expect(res.body.data).toMatchObject([
+      { id: 3, title: 'Article D', ...BASIC_ARTICLE_FIELDS },
+      { id: 1, title: 'Article A', ...BASIC_ARTICLE_FIELDS },
+      { id: 2, title: 'Article C', ...BASIC_ARTICLE_FIELDS },
+    ]);
+  });
+
+  // This one will fail because we don't support deep sort with deep filter
+  // The where and orderBy will be applied to two different joins (instead of a single one), impacting the results
+  test.skip('Deep sort (1st level) + Deep filters (1st level)', async () => {
```

Comment:
can you create a ticket for this to be supported 🙏

Reply:
Yep, done in `DX-1473`